### PR TITLE
Unit test fixes

### PR DESCRIPTION
**Pull request by bravostuzero**
Made several changes to the tests for bug fixes and to reflect best practice:
- use bash 'pipefail' mode so that the exit status reflects that of the unit tests instead of tee
- use one container per LMC device
- unit tests wait for LMC devices to be running rather than just registered
- small python 3 fix

### CODE REVIEW

File: test-harness/Makefile [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
 test:
-	retry -- tango_admin --check-device ska/basedevice/01
-	retry -- tango_admin --check-device ska/alarmhandler/01
-	retry -- tango_admin --check-device ska/capability/01
-	retry -- tango_admin --check-device ska/logger/01
-	retry -- tango_admin --check-device ska/master/01
-	retry -- tango_admin --check-device ska/obsdevice/01
-	retry -- tango_admin --check-device ska/subarray/01
-	retry -- tango_admin --check-device ska/telstate/01
+	retry --max=10 -- tango_admin --ping-device ska/basedevice/01
```
Comment (adityadangeska) [inline test-harness/Makefile]: Pipeline fails with the ping commands.

**Review by adityadangeska — CHANGES_REQUESTED** [review]
Overall the changes are fine but the pipeline fails when 'make test' is run.

**Comment by flyingfrog81** [comment]
Is this stale? I see there is a lot of value in this PR a part from the ping failing which could also be reverted. @bravostuzero , @adityadangeska can you please continue this conversation? Is this still applicable?

**Comment by bravostuzero** [comment]
The changes are still very relevant and should be merged in some form ASAP. The critical change is to test-harness/Makefile; without it, the CI run report success even when the unit tests fail.

The changes from '--check-device' to '--ping-device' are not so critical, but they're still important because they check that the device is running rather than just registered with the Tango database. If the --ping-device check fails then that is worth investigating in itself, because it indicates that the TMC device failed to start for some reason.

**Comment by bravostuzero** [comment]
What's the status of this PR? The 'bash pipefail' change need to be merged ASAP, otherwise the CI tests incorrectly report success and the CI status is useless. 

The same is true for tmc-prototype too: the bash pipefail patch MUST be added, otherwise the CI status is broken.

File: test-harness/Makefile (hunk 1 above)
Comment (bravostuzero) [reply test-harness/Makefile]: if the pipeline fails when pinging the TMC devices, then the PR is working correctly! It indicates that one or more TMC devices could not be started in container, so there's either a problem with the device code or with the container setup.

**Review by adityadangeska — APPROVED** [review]

**Pull request merged**